### PR TITLE
Integrate llvm-project @f1595ecfdce5387e41826fd72ff930a1a39ae398

### PR DESCRIPTION
**Pull request by Max191**
- bump llvm-project to https://github.com/llvm/llvm-project/commit/f1595ecfdce5387e41826fd72ff930a1a39ae398
  - revert https://github.com/iree-org/llvm-project/commit/1004865f1ca41a9581da8747f34b29862d3ebc3d due to compiler failures in VectorDistribute. Tracked in https://github.com/iree-org/iree/issues/18955.
- bump stablehlo to https://github.com/openxla/stablehlo/commit/c32f7c215d9bed2d4077a7f884781f02e61df292
- bump torch-mlir to https://github.com/llvm/torch-mlir/commit/8b0bf2e2930cc4ef0c9e1212b31c2c4fad2d9141
  - revert https://github.com/llvm/torch-mlir/commit/55ff110dc29cab7e2495ccdbec9a60512c29c665 due to new onnx failures. Tracked in https://github.com/iree-org/iree/issues/18961

### CODE REVIEW

**Comment by Max191** [comment]
Waiting on stablehlo and torch-mlir to bump llvm, and then will bump them here to fix build failures.

EDIT: Both landed and bumped.

File: tests/external/iree-test-suites/onnx_ops/onnx_ops_cpu_llvm_sync.json [hunk 1]
```diff
@@ -380,6 +379,7 @@
     "onnx/node/generated/test_sce_none_weights_expanded",
     "onnx/node/generated/test_sce_none_weights_log_prob",
     "onnx/node/generated/test_sce_none_weights_log_prob_expanded",
+    "onnx/node/generated/test_shape_clip_start",
```
Comment (Max191) [inline tests/external/iree-test-suites/onnx_ops/onnx_ops_cpu_llvm_sync.json:382]: This shape test fails on top of main too, so I don't think the failure is related to the integrate. There is an issue filed in torch-mlir for this test here: https://github.com/llvm/torch-mlir/issues/3841.

**Review by MaheshRavishankar — APPROVED** [review]

**Comment by benvanik** [comment]
@Max191 this has a typo in one of the tests (tile_and_distribute_to_workgroups.mlir):

```
        %cl_4 = hal.interface.constant.load layout(#pipeline_layout) ordinal(4) : i32
        %cl_5 = hal.interface.constant.load layout(#pipeline_layout) ordinal(5) : i32
        %0 = arith.index_castui %cl_0 : i32 to index
        %1 = arith.index_castui %cl_1 : i32 to index
        %2 = arith.index_castui %cl_2 : i32 to index
        %3 = arith.index_castui %cl_3 : i32 to index
        %tile0 = arith.index_castui %cl_3 : i32 to index
        %tile1 = arith.index_castui %cl_3 : i32 to index
```

that should be 
```
        %tile0 = arith.index_castui %cl_4 : i32 to index
        %tile1 = arith.index_castui %cl_5 : i32 to index
```
(the new ones that were added)

**Comment by Max191** [comment]
> @Max191 this has a typo in one of the tests (tile_and_distribute_to_workgroups.mlir):

Fixed here: https://github.com/iree-org/iree/pull/18982

**Comment by ScottTodd** [comment]
Build error (warning as error) on gcc: https://github.com/iree-org/iree/actions/runs/11627279570/job/32380341216#step:4:6312

**Comment by ScottTodd** [comment]
> Build error (warning as error) on gcc: https://github.com/iree-org/iree/actions/runs/11627279570/job/32380341216#step:4:6312

Left a comment upstream: https://github.com/llvm/torch-mlir/pull/3813#discussion_r1826102619

**Pull request merged**